static/js/main.js

```diff
@@ -260,14 +267,25 @@ function flipImage(screenshot_id) {
     canvas.height = img.naturalHeight;
 
     let context = canvas.getContext('2d');
-    context.scale(-1, 1);
-    context.drawImage(img, -img.naturalWidth, 0);
+    
+    let direction = ''
+    if (axis === 'y') {
+        direction = 'flipped_horizontally';
+        context.scale(-1, 1);
+        context.drawImage(img, -img.naturalWidth, 0);
+    } else if (axis === 'x') {
+        direction = 'flipped_vertically';
+        context.scale(1, -1);
+        context.drawImage(img, 0, -img.naturalHeight);
+    } else {
+        throw("Invalid flip axis.")
+    }
```

Comment:
I still think we can do better with sharing code.
You did do good pulling out some code, but the problem is that non-trivial things are sill not shared.
Both `context.scale()` and `context.drawImage()` can be tricky, so we would prefer if they were only done in one place.

What about setting variables like `scaleX`, `scaleY`, `drawX`, `drawY` according to the direction.
(You can even be a cool kid and merge it with the above comment about booleans by defaulting the variables to one value and then have a single if branch that changes for the other flip direction.)